.github/workflows/deploy-docs.yml

```diff
@@ -0,0 +1,81 @@
+name: Deploy docs on gh-pages
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build-environment:
+    runs-on: ubuntu-18.04
+    name: Build conda environment
+    steps:
+      - uses: actions/checkout@v2
+        name: Checkout repository
+
+      # See: https://github.com/marketplace/actions/setup-conda
+      - uses: s-weigand/setup-conda@v1
+        with:
+          conda-channels: "conda-forge"
+
+      # Build cache of environment -- this can shave minutes off the CI.
+      - name: Cache conda environment
+        id: cache-environment
+        uses: actions/cache@v2
+        # Conda environment build step depends on just environment.yml,
+        # so we ensure that the hash key contains its hash.
+        # If the file changes, then its hash will change,
+        # and the cache will be invalidated,
+        # thus triggering a rebuild.
+        # (There is a strong assumption here that changing `build_environment.sh`
+        # will not change the environment definition, which it shouldn't.)
+        with:
+          path: |
+            jax-unirep.tar.gz
+          key: ${{ runner.os }}-${{ hashFiles('environment.yml') }}
+
+      - name: Build and pack environment
+        if: steps.cache-environment.outputs.cache-hit != 'true'
+        run: bash scripts/ci/build_environment.sh
+
+      # See: https://github.com/actions/upload-artifact
+      - name: Upload environment
+        uses: actions/upload-artifact@v2
+        with:
+          name: jax-unirep-tarball
+          path: jax-unirep.tar.gz
+
+  build-and-deploy-docs:
+    runs-on: ubuntu-18.04
+    needs: build-environment
+    name: Build and deploy docs
+    steps:
+      - uses: actions/checkout@v2
+        name: Checkout repository
+
+      # https://github.com/actions/download-artifact
+      - name: Download environment tarball
+        uses: actions/download-artifact@v2
+        with:
+          name: jax-unirep-tarball
+
+      - name: Unpack environment and activate it
+        run: bash scripts/ci/unpack_environment.sh
```

Comment:
I think this stretch might not be necessary, i.e. no need to split the job into two. This is because there's no downstream splitting of tasks like what happens in the CI checks. That said, I think we can merge first and see what happens. If there's nothing wrong, might not be much point in changing this.

Reply:
Ah good point, so up- and downloading the env is also not necessary, right? I guess this would save some time.

Reply:
Yes, that is correct!